src/provider/systemdmanager/systemdunit.rs

```diff
@@ -131,38 +267,32 @@ impl SystemDUnit {
     }
 
     /// Add a key=value entry to the specified section
-    fn add_property(&mut self, section: &'static str, key: &str, value: &str) {
-        let section = self
-            .sections
-            .entry(String::from(section))
-            .or_insert_with(HashMap::new);
+    fn add_property(&mut self, section: Section, key: &str, value: &str) {
+        let section = self.sections.entry(section).or_insert_with(HashMap::new);
         section.insert(String::from(key), String::from(value));
     }
 
-    fn add_env_var(&mut self, name: &str, value: &str) {
-        self.environment
-            .insert(String::from(name), String::from(value));
-    }
-
     /// Retrieve content of the unit file as it should be written to disk
     pub fn get_unit_file_content(&self) -> String {
-        let mut unit_file_content = String::new();
+        SECTION_ORDER
```

Comment:
If someone added some entry to a section later on, without adding that section to the section order vec, no entries would be written for that section here. 
Should we maybe check if there are any "orphan" sections in self.sections and panic if that is the case, as this would signify a coding error?
Alternatively, the simplest solution would be to just sort by section name, I don't think systemd cares about ordering which would make this a purely cosmetic issue. And I am fully aware, that I am the guy that started down that road and dragged you along :)

Reply:
We have two options:

1. Sort by section name and ignore any file aesthetics.
2. Use a macro which creates the error-prone boilerplate, e.g. [EnumIter](https://docs.rs/strum_macros/0.20.1/strum_macros/derive.EnumIter.html). On the one hand, additional dependencies are bad, on the other hand, I use these things very often in Haskell and I don't know if I can live without it.

Reply:
For what it's worth I use EnumIter in operator-rs as well.

Reply:
I changed the code to use `EnumIter`.